java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S5122_java.html

```diff
@@ -0,0 +1,68 @@
+<ul>
+  <li> <a href="http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2018-0269">CVE-2018-0269</a> </li>
+  <li> <a href="http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2017-14460">CVE-2017-14460</a> </li>
+</ul>
+<p>Applications that enable CORS will effectively relax the same-origin policy in browsers, which is in place to prevent AJAX requests to hosts other
+than the one showing in the browser address bar. Being too permissive, CORS can potentially allow an attacker to gain access to sensitive
+information.</p>
+<p>This rule flags code that enables CORS or specifies any HTTP response headers associated with CORS. The goal is to guide security code reviews.</p>
+<h2>Ask Yourself Whether</h2>
+<ul>
+  <li> Any URLs responding with <code>Access-Control-Allow-Origin: *</code> include sensitive content. </li>
+  <li> Any domains specified in <code>Access-Control-Allow-Origin</code> headers are checked against a whitelist. </li>
+</ul>
+<h2>Recommended Secure Coding Practices</h2>
+<ul>
+  <li> The <code>Access-Control-Allow-Origin</code> header should be set only on specific URLs that require access from other domains. Don't enable
+  the header on the entire domain. </li>
+  <li> Don't rely on the <code>Origin</code> header blindly without validation as it could be spoofed by an attacker. Use a whitelist to check that
+  the <code>Origin</code> domain (including protocol) is allowed before returning it back in the <code>Access-Control-Allow-Origin</code> header.
+  </li>
+  <li> Use <code>Access-Control-Allow-Origin: *</code> only if your application absolutely requires it, for example in the case of an open/public API.
+  For such endpoints, make sure that there is no sensitive content or information included in the response. </li>
+</ul>
+<h2>Questionable Code Example</h2>
+<pre>
+// === Java Servlet ===
+@Override
+protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+  resp.setHeader("Content-Type", "text/plain; charset=utf-8");
+  resp.setHeader("Access-Control-Allow-Origin", "http://localhost:8080"); // Questionable
+  resp.setHeader("Access-Control-Allow-Credentials", "true"); // Questionable
+  resp.setHeader("Access-Control-Allow-Methods", "GET"); // Questionable
+  resp.getWriter().write("response");
+}
+</pre>
+<pre>
+// === Spring MVC Controller annotation ===
+@CrossOrigin(origins = "http://domain1.com") // Questionable
+@RequestMapping("")
+public class TestController {
+    public String home(ModelMap model) {
+        model.addAttribute("message", "ok ");
+        return "view";
+    }
+
+    @CrossOrigin(origins = "http://domain2.com") // Questionable
+    @RequestMapping(value = "/test1")
+    public ResponseEntity&lt;String&gt; test1() {
+        return ResponseEntity.ok().body("ok");
+    }
+}
+</pre>
+<h2>Exceptions</h2>
+<p>If more than one CORS related header is set per method/function, only one issue will be raised.</p>
```

Comment:
This exception does not reflect current implementation. The exception should either be removed from RSPEC or implementation updated accordingly

Reply:
RSPEC has been modified thanks to your suggestion